includes/fields/class-phone.php

```diff
@@ -64,6 +64,14 @@ public static function init( $meta = [] ) {
 						'options'     => 'countries',
 						'_order'      => 120,
 					],
+
+					'unique'    => [
```

Comment:
Let's move this option outside of the field classes since it's specific to Attributes API (values should be unique among attributes, while we try to keep Fields API as abstract as possible), add this checkbox similarly to how we add Synced and Moderated checkboxes, it should appear in the Edit section of the attribute settings meta box. You can add a list of hard-coded field types there that are ok for this option (e.g. Text, Number...).